File: iFlat/src/main/java/com/iflat/report/bean/mm/PaintMRP.java
package com.iflat.report.bean.mm;

/**
 * Created by tyriv on 2016/1/11.
 */
public class PaintMRP {
    private String projNo;
    private String code;
    private String name;
    private String description;
    private String unit;
    private double demand;  //需求数量
    private double hasRequest;  //已申请数量
    private double carryOver;  //结转数量
    private double delivery;   //出库
    private double rest;  //可申请余量

    public String getProjNo() {
        return projNo;
    }

    public void setProjNo(String projNo) {
        this.projNo = projNo;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getUnit() {
        return unit;
    }

    public void setUnit(String unit) {
        this.unit = unit;
    }

    public double getDemand() {
        return demand;
    }

    public void setDemand(double demand) {
        this.demand = demand;
    }

    public double getHasRequest() {
        return hasRequest;
    }

    public void setHasRequest(double hasRequest) {
        this.hasRequest = hasRequest;
    }

    public double getCarryOver() {
        return carryOver;
    }

    public void setCarryOver(double carryOver) {
        this.carryOver = carryOver;
    }

    public double getDelivery() {
        return delivery;
    }

    public void setDelivery(double delivery) {
        this.delivery = delivery;
    }

    public double getRest() {
        return rest;
    }

    public void setRest(double rest) {
        this.rest = rest;
    }
}

File: web/app/model/report/mm/PaintMRP.js
Ext.define('iFlat.model.report.mm.PaintMRP', {
    extend: 'Ext.data.Model',
    alias: 'model.rpt-mm-paintMRP',

    fields: [
        {name: 'projNo', mapping: 'projNo', type: 'string'},
        {name: 'code', mapping: 'code', type: 'string'},
        {name: 'name', mapping: 'name', type: 'string'},
        {name: 'description', mapping: 'description', type: 'string'},
        {name: 'unit', mapping: 'unit', type: 'string'},
        {
            name: 'demand',
            mapping: function (data) {
                return data['demand'].toFixed(2);
            },
            type: 'number'
        },{
            name: 'hasRequest',
            mapping: function (data) {
                return data['hasRequest'].toFixed(2);
            },
            type: 'number'
        },{
            name: 'carryOver',
            mapping: function (data) {
                return data['carryOver'].toFixed(2);
            },
            type: 'number'
        },{
            name: 'delivery',
            mapping: function (data) {
                return data['delivery'].toFixed(2);
            },
            type: 'number'
        },{
            name: 'rest',
            mapping: function (data) {
                return data['rest'].toFixed(2);
            },
            type: 'number'
        }
    ]
})